app/helpers/vuelidate_form/vuelidate_form_builder/vuelidate_field_builder.rb

```diff
@@ -1,61 +1,97 @@
-		@slot_scope = options.delete(:slot_scope) || SLOT_SCOPE
-		@parent_scope = options.delete(:parent_scope)
+		@options = HashWithIndifferentAccess.new(options.delete(:field_options) || {}).merge(options.extract! *FIELD_OPTIONS)
 		do_setup
 	end
 
-	def field(after_method = nil, selector=:div, &block)
+	# set up all the options needed for the field
+	def do_setup
+		get_validation
+		process_options
+	end
+
+	# Generate the standard html for the field
+	# @param after_method [Symbol] the name of a method to run on the form or template after the rest of the inner html is generated
+	# @param selector [Symbol] the selector for the field wrapper
+	# @yieldreturn [ActiveSupport::SafeBuffer] html content to be wrapped by the form field. Usually an input tag from +@formbuilder+
+	# @return [ActiveSupport::SafeBuffer] the generated html for the form field
+	def field(after_method = nil, selector = :div, &block)
 		custom_field(after_method, selector) do
 			field_inner &block
 		end
 	end
 
+	# Generate custom html for the field
+	# @param (see #field)
+	# @param (see #field)
+	# @yieldreturn [ActiveSupport::SafeBuffer] html content to be wrapped by the form field. Usually something in addition to an input tag from +@formbuilder+
 	def custom_field(after_method=nil, selector = :div, &block)
+		# add the field's value to the form unless options say not to
+		@formBuilder.add_value(@attribute) unless @options[:skip_value]
+		# grab the after method for later user
 		@after_method = after_method if after_method.present?
-
-		output = @formBuilder.step(@options[:is_step], @options.delete(:step_options) || {}) do
-			@template.content_tag(:"field-errors", @err_args) do
-				@template.content_tag(selector, @field_args, &block) <<
+		# get stepper options
+		step_options = @options.delete(:step_options) || {}
+		# build the wizard step first (it won't wrap if this isn't a step)
+		@formBuilder.step(@is_step, step_options.symbolize_keys) do
+			# wrap the field-errors component in the step
+			@template.content_tag(:"field-errors", @error_wrapper_options) do
+				# wrap the inner wrapper in the component and around the block output
+				@template.content_tag(selector, @error_inner_options, &block) <<
+				# add the error field for browsers with javascript disabled
 				no_js_error_field <<
+				# call the after method
 				call_after_method
 			end
 		end
-
-		@formBuilder.add_validation(@attribute) if @validate
-		output
 	end
 
-
+	# Generate the standard html to be wrapped in +#field+
 	def field_inner
 		output = ActiveSupport::SafeBuffer.new
-
+		#first call the given block
 		output << yield if block_given?
 
+		# add the label based on where the options say it should go
 		@options[:label_last] ? output << field_label : output.prepend(field_label)
+		# add the tooltip
 		output << field_tooltip
 	end
 
+	# generate the html for an error field for browsers with javascript disabled
```

Comment:
are u still supporting javascript being disabled?

Reply:
not as vigorously. there are some parts of the app that it would be too hard to do with, like the encounter form